### PR TITLE
FullText bloom filter index added support for Array data type

### PR DESCRIPTION
**Pull request by kitaisreal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Skip indexes `tokenbf_v1` and `ngrambf_v1` added support for Array data type with key of String of FixedString type.

### CODE REVIEW

File: src/Storages/MergeTree/MergeTreeIndexFullText.cpp [hunk 1]
```diff
@@ -378,6 +410,15 @@ bool MergeTreeConditionFullText::atomFromAST(
         else if (!token_extractor->supportLike() && (func_name == "like" || func_name == "notLike"))
             return false;
 
+        if (func_name == "has")
+        {
+            out.key_column = key_column_num;
+            out.function = RPNElement::FUNCTION_HAS;
+            out.bloom_filter = std::make_unique<BloomFilter>(params);
+            stringToBloomFilter(const_value.get<String>(), token_extractor, *out.bloom_filter);
```
Comment (vdimir) [inline src/Storages/MergeTree/MergeTreeIndexFullText.cpp:418]: What if first argument would be constant? Should we add `"has"` to `if` above where `key_arg_pos` is checked?

File: src/Storages/MergeTree/MergeTreeIndexFullText.cpp [hunk 2]
```diff
@@ -155,13 +156,40 @@ void MergeTreeIndexAggregatorFullText::update(const Block & block, size_t * pos,
...
         {
-            auto ref = column->getDataAt(*pos + i);
-            columnToBloomFilter(ref.data, ref.size, token_extractor, granule->bloom_filters[col]);
+            const auto & column_array = assert_cast<const ColumnArray &>(*column);
+            const auto & column_offsets = column_array.getOffsets();
+            const auto & column_key = column_array.getData();
+
+            for (size_t i = 0; i < rows_read; ++i)
+            {
+                size_t element_start_row = column_offsets[current_position - 1];
```
Comment (vdimir) [inline src/Storages/MergeTree/MergeTreeIndexFullText.cpp:171]: Why do we sure that `*pos` is not zero?

File: src/Storages/MergeTree/MergeTreeIndexFullText.cpp (hunk 1 above)
Comment (kitaisreal) [reply src/Storages/MergeTree/MergeTreeIndexFullText.cpp:418]: It is checked above before we get into this function, this argument const_value is constant that we get from KeyCondition.

File: src/Storages/MergeTree/MergeTreeIndexFullText.cpp (hunk 2 above)
Comment (kitaisreal) [reply src/Storages/MergeTree/MergeTreeIndexFullText.cpp:171]: It is required by contract, reference would be better but I did not want to change interfaces yet.

**Review by vdimir — APPROVED** [review]

**Pull request merged**